### PR TITLE
Copy `cortex/pkg/querier/stats` package dependency into Loki

### PR DESCRIPTION
**Pull request by DylanGuedes**
**What this PR does / why we need it**:
As a part of removing Loki dependency on cortex packages, we decided to copy the `querier/stats` package from cortex to Loki itself.
Note that, since stats has proto definitions, I had to regenerate them so it might have some diff between the Loki and Cortex version.
Also, note that I can't completely get rid of the vendored cortex `pkg/querier/stats` because Cortex itself still using it. So, this is actually an intermediate step.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by kavirajk — APPROVED** [review]
LGTM :+1:

**Review by ssncferreira — APPROVED** [review]
LGTM 🎉 
Just a small comment 🙂

File: pkg/querier/worker/frontend_processor.go [hunk 1]
```diff
@@ -6,15 +6,15 @@ import (
...
-	querier_stats "github.com/cortexproject/cortex/pkg/querier/stats"
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/grafana/dskit/backoff"
 	"github.com/weaveworks/common/httpgrpc"
 	"google.golang.org/grpc"
 
 	"github.com/grafana/loki/pkg/lokifrontend/frontend/v1/frontendv1pb"
+	"github.com/grafana/loki/pkg/querier/stats"
+	querier_stats "github.com/grafana/loki/pkg/querier/stats"
```
[the remark this answers is not in the document]
Comment (DylanGuedes) [reply pkg/querier/worker/frontend_processor.go:17]: good catch. I remember having some problem regarding a conflict with using plain `stats`, but I'll double check this usage before.

[the remark this answers is not in the document]
Comment (DylanGuedes) [reply pkg/querier/worker/frontend_processor.go]: Done!

**Review by ssncferreira — APPROVED** [review]
LGTM 🏆

**Pull request merged**